internal/util/fileop_test.go

```diff
@@ -326,33 +377,86 @@ func TestFileOpCopyBase(test *testing.T) {
 			alreadyExistsDirname,
 			"",
 		},
+		{
+			alreadyExistsDirname + "/*",
+			startingEmptyDirname,
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*.txt",
+			startingEmptyDirname,
+			"",
+		},
+		{
+			alreadyExistsDirname,
+			startingEmptyDirname,
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*",
+			"a",
+			"",
+		},
+		// This will create a directory called "a.txt".
+		{
+			alreadyExistsDirname + "/*",
+			"a.txt",
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*.txt",
+			"a",
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*.txt",
+			"a.txt",
+			"",
+		},
+		{
+			"*",
+			"a",
+			"",
+		},
 		{
 			"a",
 			"b",
-			"does not exist",
+			"Unable to find source path",
 		},
 		{
 			alreadyExistsDirname,
 			alreadyExistsFilePosixRelpath,
 			"already exists",
 		},
+		{
+			alreadyExistsDirname + "/*.txt",
+			alreadyExistsFilename,
+			"does not exist or is not a dir",
+		},
 	}
 
 	for i, testCase := range testCases {
 		rawOperation := []string{"cp", testCase.source, testCase.dest}
 
 		runFileOpExecTest(test, fmt.Sprintf("Case %d", i), rawOperation, testCase.errorSubstring, func(tempDir string) {
-			expectedSource := filepath.Join(tempDir, testCase.source)
+			expectedSourceGlob := filepath.Join(tempDir, testCase.source)
```

Comment:
This is getting pretty complex.
Maybe the globs should have their own test function or you should change this one to have a []string for the expected dirents.